src/java/org/apache/nutch/indexer/IndexerMapReduce.java

```diff
@@ -256,20 +256,19 @@ public void reduce(Text key, Iterator<NutchWritable> values,
       }
```

Comment:
dbDatum is already used above (line 240 and following) when deleting gone pages and redirects. Deletions are done if either fetchDatum or dbDatum match the status (gone resp. redirect). Why not also relax these conditions so that dbDatum is optional? fetchDatum should exist, otherwise every index job will send deletions for **all** 404s/redirects in CrawlDb including those already deleted in the rounds before.

Reply:
Fixed.